src/boot/efi/stub.c

```diff
@@ -277,11 +277,17 @@ static EFI_STATUS run(EFI_HANDLE image) {
                 mangle_stub_cmdline(cmdline);
         }
 
-        /* SMBIOS strings are measured in PCR1, so we do not re-measure these command line extensions. */
         const char *extra = smbios_find_oem_string("io.systemd.stub.kernel-cmdline-extra");
         if (extra) {
                 _cleanup_free_ char16_t *tmp = TAKE_PTR(cmdline), *extra16 = xstr8_to_16(extra);
                 cmdline = xasprintf("%ls %ls", tmp, extra16);
+
+                /* SMBIOS strings are measured in PCR1, but we also want to measure them in our specific
+                 * PCR12, as firmware-owned PCRs are very difficult to use as they'll contain unpredictable
+                 * measurements that are not under control of the machine owner. */
+                m = false;
+                (void) tpm_log_load_options(extra16, &m);
+                parameters_measured = m;
```

Comment:
```c
parameters_measured = parameteres_measured < 0 ? m : parameters_measures && m;
```

i.e. `parameters_measured` should be negative if we never were tempted to measure any parameters, true if we did measure something, or false if we didn't measure because of some reason though were asked to.

Since we have multiple things that end up in PCR 12, we need to combine the results of each such measurement like that.